mysql-test/t/mysqldump_gtid_purged.inc

```diff
@@ -0,0 +1,91 @@
+--source include/have_log_bin.inc
+--source include/have_innodb.inc
+--source include/not_embedded.inc
+--source include/have_debug_sync.inc
+--source include/count_sessions.inc
+
+# start-sql-file/finish-sql-file options of mysqldump available only in debug build
+--source include/have_debug.inc
+
+# Handling of background process
+--source include/not_windows.inc
+
+
+--echo #################################################
+--echo # Testing mysqldump with options: $MYSQLDUMP_OPTS
+--echo #################################################
+
+--let $sleep_sec = 5
+--let $start_sql_file = $MYSQL_TEST_DIR/std_data/mysqldump_gtid_purged_start.sql
+--let $finish_sql_file = $MYSQL_TEST_DIR/std_data/mysqldump_gtid_purged_finish.sql
+
+
+--connect(con1, localhost, root)
+
+--connection default
+CREATE TABLE t1 (a INT) ENGINE = InnoDB;
+INSERT INTO t1 VALUES (0);
+# Capture current gtid_executed
+--let $gtid_executed_start = `SELECT @@GLOBAL.GTID_EXECUTED`
+
+# Start mysqldump in background collecting its PID
+--let $command = $MYSQL_DUMP
+--let $command_opt = $MYSQLDUMP_OPTS test t1 --start-sql-file=$start_sql_file --finish-sql-file=$finish_sql_file
+--let $output_file = $MYSQLTEST_VARDIR/tmp/dump.sql
+--let $pid_file = $MYSQLTEST_VARDIR/tmp/mysqldump.pid
+--source include/start_proc_in_background.inc
+
+# Wait for mysqldump to reach the point just before getting gtid_executed
+SET DEBUG_SYNC = 'now WAIT_FOR before_show_gtid_executed.reached';
```

Comment:
Is it OK if `before_show_gtid_executed.reached` is signaled before (by mysqldump) we start waiting on it here?

Reply:
The point here is that mysqldump is started in async way. We need to stop it just before gtid_executed is queried, but after allpotential locks and transaction is started. So mysqldump signals that that it reached this point and waits for signal to continue.
MTR test starts mysqldump and then if mysqldump gets to the needed point before MTR test starts waiting, it will signal it. MTR when requested to wait, will just go ahead as sync point is signalled.
But if MTR gets to this point before mysqldump reaches 'before_show_gtid_executed' sync point, MTR will wait for mysqldump.

Answering the question: Yes, it is OK.

Reply:
Yes, My primary concern was `Case 1`. If you know it works, then no problem here